src/renderers/dom/shared/__tests__/ReactDOMAttribute-test.js

```diff
@@ -133,5 +133,25 @@ describe('ReactDOM unknown attribute', () => {
       );
       expectDev(console.error.calls.count()).toBe(1);
     });
+
+    it('allows camelCase unknown attributes and warns', () => {
+      spyOn(console, 'error');
+
+      var el = document.createElement('div');
+      ReactDOM.render(<div helloWorld="something" />, el);
+      expect(el.firstChild.getAttribute('helloworld')).toBe('something');
+
+      expectDev(console.error.calls.count()).toBe(1);
+      expectDev(
+        normalizeCodeLocInfo(console.error.calls.argsFor(0)[0]),
+      ).toMatch(
+        'React does not recognize the `helloWorld` prop on a DOM element. ' +
+          'If you intentionally wanted it to appear in the DOM as a custom ' +
+          'attribute, spell it as the lowercase `helloworld` instead. ' +
+          'If you passed it accidentally from a parent component, remove ' +
```

Comment:
I think it’s important to mention this since people will often get this warning due to spreading.

Reply:
It seems like practically this is going to make the unknown DOM warning permanent (modulo some cases). Granted it's probably not what the user intends it seems a bit heavy handed.

Reply:
or perhaps maybe just that it's going to give the impression that everything is fine when they aren't getting this warning, while `foo="[Object object]"` still gets passed

Reply:
The react prop name can still have casing, as long as it's passed as lowercase to the HTML element. This doesn't allow spreading of attributes, but you can do it at least.

I think it's important to communicate what this is protecting against. I don't know that this should live in the warning, but I do agree that a developer not familiar with the problem-space could see this as an arbitrary restriction. 

We could turn it into a "React's got your back" kind of thing if we effectively communicate it.

Reply:
What about custom elements (or does that not hit here)? Chance of a false positive? If we are trying to get out of folks way about passing what they want to the DOM this feels a bit arbitrary as you say.

Reply:
sigh didn't see the original issue, sorry for the noise :P